zipline/utils/sentinel.py

```diff
@@ -4,22 +4,42 @@
 Sentinel objects are used when you only care to check for object identity.
 """
 import sys
+from textwrap import dedent
 
 
 def sentinel(name, doc=None):
-    @object.__new__   # bind a single instance to the name 'NotSpecified'
-    class result(object):
+    try:
+        value = sentinel._cache[name]  # memoized
+    except KeyError:
+        pass
+    else:
+        if doc == value.__doc__:
+            return value
+
+        raise ValueError(dedent(
+            """\
+            New sentinel value %r conflicts with an existing sentinel of the
```

Comment:
Should there be a backslash here?  I feel like wrapping the exception message isn't valuable here.